src/lib/Controller/api/robot_private.h

```diff
@@ -23,11 +23,16 @@
 #include "request.h"
 #include "webots/nodes.h"
 
+#ifndef NDEBUG
 #define ROBOT_ASSERT(condition)                                                     \
   {                                                                                 \
     if (!(condition))                                                               \
       robot_abort("%s:%d: assertion failed: %s\n", __FILE__, __LINE__, #condition); \
   }
+#else
+#define ROBOT_ASSERT(condition) \
+  {}
+#endif
```

Comment:
```suggestion
#ifdef NDEBUG
#define ROBOT_ASSERT(condition) \
  {}
#else
#define ROBOT_ASSERT(condition)                                                     \
  {                                                                                 \
    if (!(condition))                                                               \
      robot_abort("%s:%d: assertion failed: %s\n", __FILE__, __LINE__, #condition); \
  }
#endif
```

Just to avoid double negation.